### PR TITLE
Improve CI logic

### PR DESCRIPTION
**Pull request by juliotrigo**
* Add Travis test `stage` and amend python Tox env
* Remove Python 3.7 logic until that version is tested by Travis:
  * PyPI 3.7 classifier 
  * py37 element from Tox env list
* Update `dev` dependencies and make Tox use them
* Simplify Tox file and make better use of its settings
* Update PyPI package classifiers

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,12 +1,9 @@
...
 
 [testenv]
 whitelist_externals = make
-deps =
-    flake8
-    coverage
-    pytest==3.2.5
 commands =
-    make coverage ARGS='-x -vv --disable-pytest-warnings'
+    pip install -U --editable ".[dev]"
```
Comment (bolsote) [inline tox.ini]: We could also use this to remove this line completely:

https://tox.readthedocs.io/en/latest/config.html#conf-extras

File: setup.py [hunk 2]
```diff
@@ -25,9 +25,9 @@
     install_requires=['nameko>=2.11,<3.0'],
     extras_require={
         'dev': [
-            'pytest==3.2.5',
-            'flake8',
-            'coverage',
+            'pytest==4.3.0',
+            'flake8>=3.7.6',
```
Comment (bolsote) [inline setup.py]: Do we need to pin this and `coverage` with such precise versions?

Comment (juliotrigo) [reply setup.py]: Not really. These are currently the latest ones but I don't think it's needed.
I'll change it back.

File: tox.ini (hunk 1 above)
Comment (juliotrigo) [reply tox.ini]: Thanks for the suggestion! I'll include both that option and `usedevelop`, as discussed.

**Review by bolsote — APPROVED** [review]

**Pull request merged**